File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode, path_max = -10_001) -> int:
        if not root:
            return 0
        return (root.val >= path_max) + \
            self.goodNodes(root.left, max(root.val, path_max)) + \
            self.goodNodes(root.right, max(root.val, path_max))

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function goodNodes(root: TreeNode | null, pathMax: number = -Infinity): number {
  if (!root) {
    return 0;
  }
  
  return (root.val >= pathMax ? 1 : 0) +
    goodNodes(root.left, Math.max(root.val, pathMax)) +
    goodNodes(root.right, Math.max(root.val, pathMax));
};